pysmt/solvers/msat.py

```diff
@@ -40,62 +37,68 @@
                               SolverStatusError,
                               InternalSolverError,
                               NonLinearError, PysmtValueError, PysmtTypeError,
-                              ConvertExpressionError)
+                              ConvertExpressionError, PysmtUnboundedOptimizationError)
 from pysmt.decorators import clear_pending_pop, catch_conversion_error
 from pysmt.solvers.qelim import QuantifierEliminator
 from pysmt.solvers.interpolation import Interpolator
 from pysmt.walkers.identitydag import IdentityDagWalker
+from pysmt.solvers.optimizer import SUAOptimizerMixin, IncrementalOptimizerMixin
 
 
-class MSatEnv(object):
+class MSatEnv():
     """A wrapper for the msat_env object.
 
     Objects within pySMT should only reference the msat_environment
     through this object. When calling a function from the underlying
     wrapper, the inner instance of msat_env needs to be used.
     This is done using the __call__ method: e.g.,
        env = MSatEnv()
-       mathsat.function(env())
+       self._msat_lib.function(env())
     """
-    __slots__ = ['msat_env']
+    __slots__ = ['msat_env', '_msat_lib']
```

Comment:
The fact that we were using slots here makes me think that there are situations in which we were creating multiple environments and trying to keep the memory consumption low. We should consider whether this is a case, as this would increase the memory usage further.